hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -832,15 +833,17 @@ public boolean isPartitionReplicaVersionStale(int partitionId, long[] versions,
 
     @Override
     public Map<Address, List<Integer>> getMemberPartitionsMap() {
-        final Collection<Member> dataMembers = node.getClusterService().getMembers(DATA_MEMBER_SELECTOR);
-        final int dataMembersSize = dataMembers.size();
+        Collection<Member> dataMembers = node.getClusterService().getMembers(DATA_MEMBER_SELECTOR);
+        int dataMembersSize = dataMembers.size();
```

Comment:
Can `dataMembersSize` be zero? If yes, we will have an array initialized with int-max capacity and that means possible OOME.